File: php/TestParcelSearchSample.php
<?php
/**
cancel the waybill
*/
include_once "EncryptUtil.php";
include_once "HttpRequestUtil.php";

$requestTime = time(); //timestamp of request
$secretKey = ""; //secretKey from Fodel
$params = array();
$params["shop_id"] = "2";//awb number
$params["app_key"] = ""; //app key from Fodel
$params["awb"] = "443351033";
$params["validate_code"] = "";
$params["ts"] = $requestTime;
//generate the sign from the paramters with eht secretKey
$signString = EncryptUtil::generateSign($params,$secretKey);
$params['sign'] = $signString;

//Set header ,add the api version to the parameter fodel_api_version
$header = array();
$header[] = "fodel_api_version: v1";
$httprequest = new HttpRequestUtil();
$httprequest->setBaseUrl("http://pos.test.fo-del.com");//host of fodel
$respone = $httprequest->post("/parcel/search",json_encode($params),true,$header);
echo $respone->body;

File: java/TestParcelSearchSample.java

import java.io.IOException;
import java.text.ParseException;
import java.util.HashMap;
import java.util.Map;
import com.alibaba.fastjson.JSONException;
import com.alibaba.fastjson.JSONObject;

import util.EncryptUtil;
import util.FodelAPIException;
import util.HttpRequestUtil;
public class TestParcelSearchSample {
	public static void main(String[] args) throws JSONException, IOException, ParseException, FodelAPIException{
		String secretKey = "";//secretKey from Fodel
        String appKey = "";//appKey from Fodel
		String requestTime = (System.currentTimeMillis()/1000)+"";
		Map<String ,String> params = new HashMap<String,String>();
		params.put("app_key", appKey);
		params.put("shop_id", "1087");
		params.put("awb", "443351033");
		params.put("validate_code", "");
		params.put("ts", requestTime);
		String signString = EncryptUtil.generateSign(params, secretKey);
		params.put("sign", signString);
		JSONObject jsonObject = HttpRequestUtil.request("GET", "/parcel/search", params);
		System.out.println(jsonObject);
	}
}
